pygmt/xarray/accessor.py

```diff
@@ -180,11 +180,9 @@ def registration(self) -> GridRegistration:
     def registration(self, value: GridRegistration | int):
         # TODO(Python>=3.12): Simplify to `if value not in GridRegistration`.
         if value not in GridRegistration.__members__.values():
-            msg = (
-                f"Invalid grid registration: '{value}'. Should be either "
-                "GridRegistration.GRIDLINE (0) or GridRegistration.PIXEL (1)."
+            raise GMTValueError(
+                value, description="grid registration", choices=GridRegistration
```

Comment:
Highlighting this for the above comment, to check that enums like `GridRegistration` are formatted properly. I.e. the error should still print "GridRegistration.GRIDLINE (0) or GridRegistration.PIXEL (1)" per https://github.com/GenericMappingTools/pygmt/pull/3696#discussion_r1948299336